packages/kit/src/runtime/app/forms.js

```diff
@@ -91,7 +101,7 @@ export function enhance(form, submit = () => {}) {
 				signal: controller.signal
 			});
 
-			result = await response.json();
+			result = deserialize(await response.text());
```

Comment:
We could also make it `deserialize(await response.json())` but I somehow felt that passing in the raw text is better.